woocommerce/woocommerce-ios
# Experiments framework: add a build script to copy GoogleService-Info.plist from mobile secrets to built directory

Part of #7397
Preparation for Firebase integration for AB testing
Platform request: pdnsEh-wr-p2


### Description


In order to integrate Firebase to the Experiments framework for AB testing, a `GoogleService-Info.plist` is required to configure Firebase on app launch. Since this plist file contains confidential information, we want to copy the file from the mobile secrets repo similar to the setup in WCAndroid.

Please excuse the diffs on the ABTest file move, it was a simple folder reorganization but I can reset those diffs manually if it makes review easier.

### Testing instructions


Feel free to test the initial integration from https://github.com/woocommerce/woocommerce-ios/pull/7398:

- Check out the branch in https://github.com/woocommerce/woocommerce-ios/pull/7398
- Place the `GoogleService-Info.plist` file in pdnsEh-wr-p2 to `.configure/woocommerce-ios/secrets` directory on your local machine; if the file has been added to mobile-secrets, try pulling the latest changes in the mobile-secrets repo and then run `bundle exec fastlane run configure_apply` in WCiOS to get the plist
- Reinstall the app if needed so that login onboarding is shown
- Launch the app and log out if needed --> a login onboarding screen should be shown, and a message should be logged in the console: `Login onboarding variation: control` or `Login onboarding variation: treatment(Optional("survey"))` along with other Firebase configuration messages



---
